src/internal/routing-table-getter/routing-table-factory.js

```diff
@@ -0,0 +1,210 @@
+      // different session towards a different router
+      return null
+    }
+  }
+}
+
+/**
+ * Create a valid routing table from a raw object
+ *
+ * @param {string} database the database name. It is used for logging purposes
+ * @param {ServerAddress} routerAddress The router address, it is used for loggin purposes
+ * @param {RawRoutingTable} rawRoutingTable Method used to get the raw routing table to be processed
+ * @param {RoutingTable} The valid Routing Table
+ */
+export function createValidRoutingTable (
+  database,
+  routerAddress,
+  rawRoutingTable
+) {
+  const expirationTime = parseTtl(rawRoutingTable, routerAddress)
+  const { routers, readers, writers } = parseServers(
+    rawRoutingTable,
+    routerAddress
+  )
+
+  assertNonEmpty(routers, 'routers', routerAddress)
+  assertNonEmpty(readers, 'readers', routerAddress)
+
+  return new RoutingTable({
+    database,
+    routers,
+    readers,
+    writers,
+    expirationTime
+  })
+}
+
+/**
+ * Parse server from the RawRoutingTable.
+ *
+ * @param {RawRoutingTable} rawRoutingTable the raw routing table
+ * @param {string} routerAddress the router address
+ * @returns {Object} The object with the list of routers, readers and writers
+ */
+function parseServers (rawRoutingTable, routerAddress) {
+  try {
+    let routers = []
+    let readers = []
+    let writers = []
+
+    rawRoutingTable.servers.forEach(server => {
+      const role = server.role
+      const addresses = server.addresses
+
+      if (role === 'ROUTE') {
+        routers = parseArray(addresses).map(address =>
+          ServerAddress.fromUrl(address)
+        )
+      } else if (role === 'WRITE') {
+        writers = parseArray(addresses).map(address =>
+          ServerAddress.fromUrl(address)
+        )
+      } else if (role === 'READ') {
+        readers = parseArray(addresses).map(address =>
+          ServerAddress.fromUrl(address)
+        )
+      } else {
+        throw newError('Unknown server role "' + role + '"', PROTOCOL_ERROR)
+      }
+    })
+
+    return {
+      routers: routers,
+      readers: readers,
+      writers: writers
+    }
+  } catch (error) {
+    throw newError(
+      `Unable to parse servers entry from router ${routerAddress} from addresses:\n${JSON.stringify(
+        rawRoutingTable.servers
+      )}\nError message: ${error.message}`,
+      PROTOCOL_ERROR
+    )
+  }
+}
+
+/**
+ * Parse the ttls from the raw routing table and return it
+ *
+ * @param {RawRoutingTable} rawRoutingTable the routing table
+ * @param {string} routerAddress the router address
+ * @returns {number} the ttl
+ */
+function parseTtl (rawRoutingTable, routerAddress) {
+  try {
+    const now = int(Date.now())
+    const expires = int(rawRoutingTable.ttl)
+      .multiply(1000)
+      .add(now)
+    // if the server uses a really big expire time like Long.MAX_VALUE this may have overflowed
+    if (expires.lessThan(now)) {
```

Comment:
Looks a bit dangerous, maybe you can do the calculations in seconds instead to get read of the multiplication and potential overflow.

Reply:
I already have test which considers the overflow and it puts the `Integer.MAX_VALUE` as the expiration time. It is the current behaviour of the routing table expiration time nowadays.